src/backends/beamcoder.ts

```diff
@@ -142,717 +88,365 @@ const getFilter = async({
     });
 };
 
+const STREAM_TYPE_VIDEO = 'video';
+
 /**
- * Class to keep track of the muxer/demuxer/encoder/muxer used while extracting a video to frames.
+ * A simple extractor that uses beamcoder to extract frames from a video file.
  */
 export class BeamcoderExtractor extends BaseExtractor implements Extractor {
```

Comment:
@animanathome as a fun history piece, all of this was previously a bunch of functions and I later refactored as a class to avoid passing so many things around all the time. Functional programming is nice until you have to pass around too many parameters everywhere.